File: spring-boot-entity-example/src/main/java/io/beaniejoy/entityexample/entity/member/Member.java
package io.beaniejoy.entityexample.entity.member;

import io.beaniejoy.entityexample.entity.review.Review;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.util.List;

@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "member")
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String email;

    private String phoneNumber;

    private Role role;

    @OneToMany(mappedBy = "member", fetch = FetchType.LAZY)
    private List<Review> reviews;
}

File: spring-test-fixtures/domain/src/main/kotlin/io/beaniejoy/entity/Member.kt
package io.beaniejoy.entity

import jakarta.persistence.*

@Entity
@Table(name = "member")
class Member(
    email: String,
    password: String,
    address: String?,
    phoneNumber: String?
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "member_id", nullable = false)
    val id: Long = 0L

    @Column(name = "email", nullable = false)
    var email: String = email
        protected set

    @Column(name = "password", nullable = false)
    var password: String = password
        protected set

    @Column(name = "address")
    var address: String? = address
        protected set

    @Column(name = "phone_number")
    var phoneNumber: String? = phoneNumber
        protected set

    companion object {
        fun createEntity(
            email: String,
            password: String,
            address: String?,
            phoneNumber: String?
        ): Member {
            return Member(
                email = email,
                password = password,
                address = address,
                phoneNumber = phoneNumber
            )
        }
    }
}